kolibri/plugins/setup_wizard/api.py

```diff
@@ -40,131 +41,24 @@ class SetupWizardResource(ViewSet):
-    def createonmyownuser(self, request):
-        """
-        Creating a Learner who has signed up using the On My Own flow
-        NOTE: This is for use when the App Plugin is *disabled*
-        """
-        facility_name = request.data.get("facility_name", "Personal")
-
-        if Facility.objects.count() == 0:
-            facility = Facility.objects.create(name=facility_name)
-        else:
-            facility = Facility.objects.get()
-            if facility_name:
-                facility.name = facility_name
-                facility.save()
-
-        if not facility:
-            return Response("facility_name is a required field", status=400)
-
-        username = request.data.get("username")
-        password = request.data.get("password")
-        full_name = request.data.get("full_name")
-
-        if not (username and password):
-            return Response(
-                "username and password fields are required fields", status=400
-            )
-
-        # Here we only expect and accept the on_my_own_setup extra_field and set it directly
-        # using the setter method for `on_my_own_setup` on Facility
-        extra_fields = request.data.get("extra_fields", None)
-        if extra_fields and extra_fields.get("on_my_own_setup"):
-            facility.on_my_own_setup = extra_fields.get("on_my_own_setup")
-
-        if extra_fields and extra_fields.get("os_user"):
-            osuser = FacilityUser.objects.get_or_create_os_user(
-                facility=facility, auth_token=request.data.get("auth_token")
-            )
-            return Response({"username": osuser.get("username")})
-
-        try:
-            user = FacilityUser.objects.create(
-                username=username, facility=facility, full_name=full_name
-            )
-            user.set_password(password)
-            user.save()
-            return Response(status=201)
-        except Exception as e:
-            return Response("Failed to create a user: {}".format(e), status=400)
-
-    @decorators.action(methods=["post"], detail=False)
-    def createsuperuser(self, request):
-        """
-        Given a username, full name and password, create a superuser attached
-        to the facility that was imported (or create a facility with given facility_name)
-        """
-        # TODO Default facility name?
-        facility_name = request.data.get("facility_name", "Personal")
-
-        # Get the imported facility (assuming its the only one at this point)
-        the_facility = Facility.get_or_create(facility_name=facility_name)
-        try:
-            superuser = FacilityUser.objects.create_superuser(
-                request.data.get("username"),
-                request.data.get("password"),
-                facility=the_facility,
-                full_name=request.data.get("full_name"),
-            )
-            return Response({"username": superuser.username})
+    def createuseronremote(self, request):
```

Comment:
@jredrejo did an interesting thing for this in merge user flow, where this was done during the validation of the task for user merging - so that if this failed for some reason there was instant feedback. Wondering if this might be an approach we could take here too.